internal/integration/rhp/v4/rhp4_test.go

```diff
@@ -0,0 +1,1383 @@
+		data := frand.Bytes(1024)
+
+		// store the sector
+		writeResult, err := rhp4.RPCWriteSector(context.Background(), transport, settings.Prices, token, bytes.NewReader(data), uint64(len(data)))
+		if err != nil {
+			t.Fatal(err)
+		}
+		roots = append(roots, writeResult.Root)
+
+		appendResult, err := rhp4.RPCAppendSectors(context.Background(), transport, cs, settings.Prices, renterKey, revision, []types.Hash256{writeResult.Root})
+		if err != nil {
+			t.Fatal(err)
+		}
+		revision.Revision = appendResult.Revision
+		checkRoots(t, roots)
+	}
+}
+
+func BenchmarkWrite(b *testing.B) {
+	n, genesis := testutil.V2Network()
+	hostKey, renterKey := types.GeneratePrivateKey(), types.GeneratePrivateKey()
+
+	hn := testutil.NewHostNode(b, hostKey, n, genesis, zap.NewNop())
+	cm := hn.Chain
+	w := hn.Wallet
+
+	results := make(chan error, 1)
+	if _, err := hn.Volumes.AddVolume(context.Background(), filepath.Join(b.TempDir(), "test.dat"), uint64(b.N), results); err != nil {
+		b.Fatal(err)
+	} else if err := <-results; err != nil {
+		b.Fatal(err)
+	}
+
+	testutil.MineAndSync(b, hn, w.Address(), int(n.MaturityDelay+20))
+
+	transport := testRenterHostPair(b, hostKey, hn, zap.NewNop())
+
+	settings, err := rhp4.RPCSettings(context.Background(), transport)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	fundAndSign := &fundAndSign{w, renterKey}
+	renterAllowance, hostCollateral := types.Siacoins(100), types.Siacoins(200)
+	formResult, err := rhp4.RPCFormContract(context.Background(), transport, cm, fundAndSign, cm.TipState(), settings.Prices, hostKey.PublicKey(), settings.WalletAddress, proto4.RPCFormContractParams{
+		RenterPublicKey: renterKey.PublicKey(),
+		RenterAddress:   w.Address(),
+		Allowance:       renterAllowance,
+		Collateral:      hostCollateral,
+		ProofHeight:     cm.Tip().Height + 50,
+	})
+	if err != nil {
+		b.Fatal(err)
+	}
+	revision := formResult.Contract
+
+	// fund an account
+	cs := cm.TipState()
+	account := proto4.Account(renterKey.PublicKey())
+	accountFundAmount := types.Siacoins(25)
+	fundResult, err := rhp4.RPCFundAccounts(context.Background(), transport, cs, renterKey, revision, []proto4.AccountDeposit{
+		{Account: account, Amount: accountFundAmount},
+	})
+	if err != nil {
+		b.Fatal(err)
+	}
+	revision.Revision = fundResult.Revision
+
+	token := proto4.AccountToken{
+		Account:    account,
+		ValidUntil: time.Now().Add(time.Hour),
+	}
+	token.Signature = renterKey.SignHash(token.SigHash())
+
+	var sectors [][proto4.SectorSize]byte
+	for i := 0; i < b.N; i++ {
+		var sector [proto4.SectorSize]byte
+		frand.Read(sector[:256])
+		sectors = append(sectors, sector)
+	}
+
+	b.ResetTimer()
+	b.ReportAllocs()
+	b.SetBytes(proto4.SectorSize)
+
+	for i := 0; i < b.N; i++ {
+		// store the sector
+		_, err := rhp4.RPCWriteSector(context.Background(), transport, settings.Prices, token, bytes.NewReader(sectors[i][:]), proto4.SectorSize)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func BenchmarkRead(b *testing.B) {
```

Comment:
`nit` makes sense to add whatever result you get here on your machine

Ran them on mine in case you want to just grab that one:
```
goos: darwin
goarch: arm64
pkg: go.sia.tech/hostd/internal/integration/rhp/v4
cpu: Apple M1 Max
=== RUN   BenchmarkRead
BenchmarkRead
BenchmarkRead-10              30          38006258 ns/op         110.36 MB/s     4495940 B/op       2524 allocs/op
PASS
ok      go.sia.tech/hostd/internal/integration/rhp/v4   5.570s
```